File: js/emit_indices.js
function
emit_indices(u, p)
{
	emit_roman_text(u, "[");

	p = cdr(p);

	if (iscons(p)) {
		emit_expr(u, car(p));
		p = cdr(p);
		while (iscons(p)) {
			emit_roman_text(u, ",");
			emit_expr(u, car(p));
			p = cdr(p);
		}
	}

	emit_roman_text(u, "]");
}

File: app/emit_indices.c
#include "app.h"

void
emit_indices(struct atom *p)
{
	emit_roman_char('[');

	p = cdr(p);

	if (iscons(p)) {
		emit_expr(car(p));
		p = cdr(p);
		while (iscons(p)) {
			emit_roman_char(',');
			emit_expr(car(p));
			p = cdr(p);
		}
	}

	emit_roman_char(']');
}
